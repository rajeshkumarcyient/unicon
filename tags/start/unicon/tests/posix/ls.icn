
$include "posix.icn"
link printf

procedure main()
   chdir("test-dir")
   f := open(".") | stop(sys_errstr(&errno), image("test-dir"))
   L := list()
   while line := read(f) do
      push(L, line)
   every write(format(stat(n := !sort(L)), n))
end

procedure format(p, name)
   s := sprintf("%4s %s %3s %8s %s",
	   p.blocks, p.mode, p.nlink,
	   p.size, name)
   
   s ||:= " -> " || \p.symlink

   return s
end

