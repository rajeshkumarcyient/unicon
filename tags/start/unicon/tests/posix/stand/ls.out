   8 drwxr-xr-x   2     4096 .
   8 drwxr-xr-x   5     4096 ..
   0 -rw-r--r--   1        0 a
   0 -rw-r--r--   1        0 b
   0 -rw-r--r--   1        0 c
   0 lrwxrwxrwx   1        1 d -> a
